code-challenges.test.js

```diff
@@ -18,40 +18,120 @@
+  test("returns an array without the first element and than shuffled", () => {
+      
+    console.log('color1 array -> after test() before expect():' ,colors1)
+      
+    expect(removeFirstAndShuffle(colors1)).toContain("yellow", "blue", "pink", "green") // here we are checking if the result array after running through our removeFirstAndShuffle function contains all of the elements from the original array minus the element at index 0. --> .toContain - Jest method - checks if it contains the data that we pass in. In this case we need to pass in the strings without being inside of an array. I think that this is because istead of checking if the result array has the strings, it checks if the result contains an array with the specific array of strings that we pass in. 
+    expect(removeFirstAndShuffle(colors1)).not.toEqual(["blue", "green", "yellow", "pink"]) // comparing the result with the original  array without the first element And NOT SHUFFLED. --> passed because the arrays should not be the same -- the result is shuffled -->  used the .not.toEaqual
+
+    expect(removeFirstAndShuffle(colors2)).toContain("saffron", "aquamarine", "periwinkle", "indigo", "ochre")
+      
+    expect(removeFirstAndShuffle(colors2)).not.toEqual(["indigo", "periwinkle", "ochre", "aquamarine", "saffron"])
+      
+    console.log('!!!!color1 array:' ,colors1)
+      
+    expect(colors1.length - 1).toEqual(["yellow", "blue", "pink", "green"].length) //here i am trying to check for the length of the newarray. if the length is equals to the length of the previous array - 1. There will be nothing else that needs to be checked.
+    expect(colors2.length - 1).toEqual(["saffron", "aquamarine", "periwinkle", "indigo", "ochre"].length)
+    
+  })
+})
+
+// !!! -> Every time I was running the test it was invoke the funciton iside of the test method changed the value of the original array (color1). Try to change the funciton! -->> this was happening because I was using .shift() to change the value of the array. I replaced .shift with slice() and assigned the new value to a variale. Than randomized the value in the variable and returned the result.
 
 // b) Create the function that makes the test pass.
-
-
+// create a function
+// use .shift() built in array method to remove the first element in the array !!-- Replaced with .slice() and assigned the result to a variable.
+// use .sort(() => Math.random() - 0.5) ---> to shuffle the array
+// return the new shorter and shuffled array
+
+const removeFirstAndShuffle = (arr) => {
+  // arr.shift()
+  let newArr = arr.slice(1)
+  // console.log(newArr)
+  console.log(newArr.sort(() => Math.random() - 0.5))
+  return newArr.sort(() => Math.random() - 0.5)
```

Comment:
Great use of the sort method and the call back function to set up your randomization